pkg/util/fedinformer/genericmanager/single-cluster-manager.go

```diff
@@ -122,7 +131,12 @@ func (s *singleClusterInformerManagerImpl) IsHandlerExist(resource schema.GroupV
 }
 
 func (s *singleClusterInformerManagerImpl) Lister(resource schema.GroupVersionResource) cache.GenericLister {
-	return s.informerFactory.ForResource(resource).Lister()
+	// Add a RLock to make sure the transform func is set before starting
+	s.lock.RLock()
```

Comment:
Do we really need lock here？ I think it can be ensured without lock.

Reply:
Given `Lister/ForResource` and `Start` are called in multiple controllers, there might be a race condition:
|  goroutine1   | goroutine2  |
|  ----  | ----  |
| ForResource  |  |
|   | Start (the informer has been created by goroutine1) |
| SetTransform (Failed since the informer has started)  |  |

Adding a lock is to make this process atomic